pre_commit/languages/python.py

```diff
@@ -31,11 +31,13 @@ def in_env(repo_cmd_runner, language_version):
 
 
 def norm_version(version):
+    version = os.path.expanduser(version)
     if os.name == 'nt':  # pragma: no cover (windows)
         if not distutils.spawn.find_executable(version):
             # The default place for python on windows is:
             # C:\PythonXX\python.exe
             version = r'C:\{0}\python.exe'.format(version.replace('.', ''))
+            version = version.replace(r'C:\\', 'C:\\')
```

Comment:
Ah, maybe remove the C:\ part entirely from the format string?  iirc \foo\bar is the same as C:\foo\bar anyway and then pre-commit cares less about what your default drive is